src/kernel/include/pgtbl.h

```diff
@@ -20,33 +20,46 @@
 #include "chal.h"
 #endif
 
+/* PRY:MODIFIED THIS FOR CORTEX-M */
 typedef enum {
 	PGTBL_PRESENT      = 1,
-	PGTBL_WRITABLE     = 1<<1,
-	PGTBL_USER         = 1<<2,
-	PGTBL_WT           = 1<<3, 	/* write-through caching */
-	PGTBL_NOCACHE      = 1<<4, 	/* caching disabled */
-	PGTBL_ACCESSED     = 1<<5,
-	PGTBL_MODIFIED     = 1<<6,
-	PGTBL_SUPER        = 1<<7, 	/* super-page (4MB on x86-32) */
-	PGTBL_GLOBAL       = 1<<8,
+	PGTBL_LEAF         = 1<<1,
+	PGTBL_WRITABLE     = 1<<2,
+	PGTBL_USER         = 1<<3,
+	PGTBL_WT           = 1<<4, 	/* write-through caching */
+	PGTBL_NOCACHE      = 1<<5, 	/* caching disabled */
 	/* Composite defined bits next*/
-	PGTBL_COSFRAME     = 1<<9,
-	PGTBL_COSKMEM      = 1<<10,     /* page activated as kernel object */
-	PGTBL_QUIESCENCE   = 1<<11,
+	PGTBL_COSFRAME     = 1<<6,
+	PGTBL_COSKMEM      = 1<<7,     /* page activated as kernel object */
 	/* Flag bits done. */
 
-	PGTBL_USER_DEF     = PGTBL_PRESENT|PGTBL_USER|PGTBL_ACCESSED|
-	                     PGTBL_MODIFIED|PGTBL_WRITABLE,
-	PGTBL_INTERN_DEF   = PGTBL_USER_DEF,
+	PGTBL_USER_DEF     = PGTBL_PRESENT|PGTBL_LEAF|PGTBL_USER|PGTBL_WRITABLE|PGTBL_WT|PGTBL_COSFRAME,
+	PGTBL_INTERN_DEF   = PGTBL_PRESENT,
 } pgtbl_flags_t;
 
-#define PGTBL_PAGEIDX_SHIFT (12)
+/* PRY:MODIFIED THIS FOR CORTEX-M */
+#define PGTBL_PAGEIDX_SHIFT (24)
```

Comment:
so you're making frames huge by default?  Not certain this will work, but I'll keep reading.